vignettes/clinical_trials.Rmd

```diff
@@ -788,3 +789,134 @@ rsp_tbl4
+}
+```
+
+This function generates counts and fractions of unique subjects corresponding to each factor level, excluding missing values (uncensored patients).
+
+A Cox proportional-hazards (Cox P-H) analysis is generated next with a third custom analysis function, `a_cph`. Prior to creating the analysis function, the Cox P-H model is fit to our data using the `coxph()` and `Surv()` functions from the `survival` package. Then this model is used as input to the `a_cph` analysis function which returns hazard ratios, 95% confidence intervals, and p-values comparing against the reference group - in this case the leftmost column.
+
+```{r}
+cph <- coxph(Surv(AVAL, CNSR == 0) ~ ACTARM + STRATA1, ties = "exact", data = adtte)
+
+a_cph <- function(df, .var, .in_ref_col, .ref_full, full_cox_fit) {
+    if (.in_ref_col) {
+        ret <- replicate(3, list(rcell(NULL)))
+    } else {
+        curtrt <- df[[.var]][1]
+        coefs <- coef(full_cox_fit)
+        sel_pos <- grep(curtrt, names(coefs), fixed = TRUE)
+        hrval <- exp(coefs[sel_pos])
+        sdf <- survdiff(Surv(AVAL, CNSR == 0) ~ ACTARM + STRATA1,
+                        data = rbind(df, .ref_full))
+        pval <- (1 - pchisq(sdf$chisq, length(sdf$n) - 1)) / 2
+        ci_val <- exp(unlist(confint(full_cox_fit)[sel_pos, ]))
+        ret <- list(rcell(hrval, format = "xx.x"),
+                    rcell(ci_val, format = "(xx.x, xx.x)"),
+                    rcell(pval, format = "x.xxxx | (<0.0001)"))
+    }
+    in_rows(.list = ret, .names = c("Hazard ratio",
+                                    "95% confidence interval",
+                                    "p-value (one-sided stratified log rank)"))
+}
+```
+
+The fourth and final analysis function, `a_tte`, generates a time to first adverse event table with three rows corresponding to Median, 95% Confidence Interval, and Min Max respectively. First a survival table is constructed from the summary table of a survival model using the `survfit()` and `Surv()` functions from the `survival` package. This table is then given as input to `a_tte` which produces the table of time to first adverse event consisting of the previously mentioned summary statistics.
+
+```{r}
+surv_tbl <- as.data.frame(summary(survfit(Surv(AVAL, CNSR == 0) ~ ACTARM,
+                                          data = adtte, conf.type = "log-log"))$table) %>%
+    dplyr::mutate(ACTARM = factor(gsub("ACTARM=", "", row.names(.)),
+                                  levels = levels(adtte$ACTARM)),
+                  ind = FALSE)
+
+a_tte <- function(df, .var,  kp_table) {
+    ind <- grep(df[[.var]][1], row.names(kp_table), fixed = TRUE)
+    minmax <- range(df[["AVAL"]])
+    mm_val_str <- format_value(minmax, format = "xx.x, xx.x")
+    rowfn <- list()
+    if (all(df$CNSR[df$AVAL == minmax[2]])) {
+        mm_val_str <- paste0(mm_val_str, "*")
+        rowfn <- "* indicates censoring"
+    }
```

Comment:
@gmbecker this is where the missing footnote is not created - it seems like there is no censoring applied in `ex_adaette`. I suppose this could be done manually if necessary. 

I compared the latest versions of `ex_adaette` and `cadaette` - `ex_adaette` is a 1200 row by 42 col dataset while `cadaette` is 3600 by 67. It is likely that the data in `formatters` is just out of date since there have been many updates in `random.cdisc.data` this year.